### DISCUSSION TITLE
Supabase OAuth session

### DISCUSSION THREAD

**Discussion by AndonMitev** [supabase/supabase#29843] [Questions] [answered]
Hello i'm successfully connect with oauth and creating supabase session and everything, just 1 thing i see `access_token` has expiration of 1 hour and `refresh_token` has no expiration but can be used only once, so do i need manually somehow refresh session with refresh token or supabase will do it automatically

**Comment by GaryAustin1** [comment, marked as answer]
With their default setup, the Supabase REST clients will handle refreshing with a timer or getting a new access_token with the refresh_token if expired.
